core/domain/html_validation_service.py

```diff
@@ -820,54 +818,149 @@ def regenerate_image_filename_using_dimensions(filename, height, width):
     return new_filename
 
 
-def add_dimensions_to_image_tags(exp_id, html_string):
+def add_dimensions_to_image_tags(file_system, html_string):
     """Adds dimensions to all oppia-noninteractive-image tags. Removes image
     tags that have no filepath.
 
     Args:
-        exp_id: str. Exploration id.
+        file_system: GcsFileSystem. File system of the entity.
         html_string: str. HTML string to modify.
 
     Returns:
         str. Updated HTML string with the dimensions for all
             oppia-noninteractive-image tags.
     """
     soup = bs4.BeautifulSoup(html_string.encode('utf-8'), 'html.parser')
+
     for image in soup.findAll(name='oppia-noninteractive-image'):
         if (not image.has_attr('filepath-with-value') or
-                image['filepath-with-value'] == ''):
+                image['filepath-with-value'] == '' or
+                image['filepath-with-value'] == '&quot;&quot;'):
             image.decompose()
             continue
 
         try:
-            filename = json.loads(unescape_html(image['filepath-with-value']))
+            filename = json.loads(
+                unescape_html(image['filepath-with-value'].replace('\\"', '')))
             image['filepath-with-value'] = escape_html(json.dumps(
-                get_filename_with_dimensions(filename, exp_id)))
+                get_filename_with_dimensions(file_system, filename)))
         except Exception as e:
+            entity_type = file_system.impl.assets_path.split('/')[0]
```

Comment:
I think you can do `(entity_type, entity_id) = file_system.impl.assets_path.split('/')` -- maybe?